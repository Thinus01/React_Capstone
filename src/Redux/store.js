import { configureStore } from '@reduxjs/toolkit';
import cryptoReducer from './data';

const store = configureStore({
  reducer: {
    crypto: cryptoReducer,
  },
});

export default store;
