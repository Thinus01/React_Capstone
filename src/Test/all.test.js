import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../Redux/store';
import Nav from '../Components/navbar';
import Models from '../Components/render';
import Crypto from '../Components/crypto';
import cryptoReducer, { fetchFinanceData } from '../Redux/data';

const crypto = { FULLName: 'bitcoin', img: 'new' };
jest.mock('axios');

describe('fetchFinanceData', () => {
  it('should fetch crypto data successfully', async () => {
    const result = await fetchFinanceData();
    expect(result).toBeDefined();
    expect(result.length).toBeGreaterThan(0);
  });
});

describe('Nav component', () => {
  it('renders Nav component', () => {
    const r = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <Nav />
        </BrowserRouter>
      </Provider>,
    ).toJSON();
    expect(r).toMatchSnapshot();
  });
  it('renders Models component', () => {
    const m = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <Models />
        </BrowserRouter>
      </Provider>,
    ).toJSON();
    expect(m).toMatchSnapshot();
  });
  it('renders Crypto component', () => {
    const m = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <Crypto crypto={crypto} />
        </BrowserRouter>
      </Provider>,
    ).toJSON();
    expect(m).toMatchSnapshot();
  });
});

describe('cryptoReducer', () => {
  it('should return the initial state', () => {
    expect(cryptoReducer(undefined, {})).toEqual([]);
  });

  it('should handle FINANCIAL_MODELING/fulfilled action', () => {
    const prevState = [{ id: 1, Name: 'Bitcoin', price: 45000 }];
    const action = {
      type: 'FINANCIAL_MODELING/fulfilled',
      payload: [
        { id: 2, Name: 'Ethereum', price: 1500 },
        { id: 3, Name: 'Litecoin', price: 180 },
      ],
    };
    const nextState = [
      { id: 1, Name: 'Bitcoin', price: 45000 },
      { id: 2, Name: 'Ethereum', price: 1500 },
      { id: 3, Name: 'Litecoin', price: 180 },
    ];
    expect(cryptoReducer(prevState, action)).toEqual(nextState);
  });
});
