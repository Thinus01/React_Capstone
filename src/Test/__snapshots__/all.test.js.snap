// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component renders Crypto component 1`] = `
<div
  className=" CryptoList"
>
  <div>
    <p
      className="Font"
    >
       
    </p>
    <div
      className="Currency"
    >
      <img
        alt="symbol"
        className="images"
        src="https://www.cryptocompare.comnew "
      />
    </div>
    <p />
  </div>
</div>
`;

exports[`Nav component renders Models component 1`] = `
Array [
  <form
    className="mb-5 nav"
  >
    <input
      aria-label="search"
      className="InputField"
      onChange={[Function]}
      placeholder="Crypto Currency"
      type="text"
      value=""
    />
  </form>,
  <div
    className="BG"
  >
    <div
      className="grid1"
    />
  </div>,
]
`;

exports[`Nav component renders Nav component 1`] = `
<nav
  className="nav"
>
  <ul
    className="Items"
  >
    <li
      className="Fonts"
    >
      <a
        aria-current="page"
        className="text-decoration-none active"
        href="/"
        onClick={[Function]}
      >
        <h3
          className="Fonts"
        >
           â®Œ 
        </h3>
      </a>
      <h5
        className="mt-1 Fonts"
      >
        Crypto Trades
      </h5>
    </li>
    <li
      className="Icons"
    >
      <h1>
        ðŸŽ¤
      </h1>
      <h1
        className="px-2"
      >
        âš™ï¸Ž
      </h1>
    </li>
  </ul>
</nav>
`;
